staging/src/k8s.io/apiserver/pkg/util/flowcontrol/request/config.go

```diff
@@ -0,0 +1,82 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package request
+
+import (
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// WorkEstimatorConfig holds work estimator parameters.
+type WorkEstimatorConfig struct {
+	ListWorkEstimatorConfig     `json:"listWorkEstimatorConfig,omitempty"`
+	MutatingWorkEstimatorConfig `json:"mutatingWorkEstimatorConfig,omitempty"`
+
+	// MinimumSeats is the minimum number of seats a request must occupy.
+	MinimumSeats uint `json:"minimumSeats,omitempty"`
+	// MaximumSeats is the maximum number of seats a request can occupy
+	//
+	// NOTE: work_estimate_seats_samples metric uses the value of maximumSeats
+	// as the upper bound, so when we change maximumSeats we should also
+	// update the buckets of the metric.
+	MaximumSeats uint `json:"maximumSeats,omitempty"`
+}
+
+// ListWorkEstimatorConfig holds work estimator parameters related to list requests.
+type ListWorkEstimatorConfig struct {
+	ObjectsPerSeat int `json:"objectsPerSeat,omitempty"`
+}
+
+// MutatingWorkEstimatorConfig holds work estimator
+// parameters related to watches of mutating objects.
+type MutatingWorkEstimatorConfig struct {
+	// TODO(wojtekt): Remove it once we tune the algorithm to not fail
+	// scalability tests.
+	Enabled                 bool            `json:"enable,omitempty"`
+	EventAdditionalDuration metav1.Duration `json:"eventAdditionalDurationMs,omitempty"`
+	WatchesPerSeat          int             `json:"watchesPerSeat,omitempty"`
+}
+
+// NewWorkEstimatorConfig creates a new WorkEstimatorConfig with default values.
+func NewWorkEstimatorConfig() WorkEstimatorConfig {
+	return WorkEstimatorConfig{
+		MinimumSeats:                1,
+		MaximumSeats:                10,
+		ListWorkEstimatorConfig:     newListWorkEstimatorConfig(),
+		MutatingWorkEstimatorConfig: newMutatingWorkEstimatorConfig(),
+	}
+}
+
+// newListWorkEstimatorConfig creates a new ListWorkEstimatorConfig with default values.
+func newListWorkEstimatorConfig() ListWorkEstimatorConfig {
+	return ListWorkEstimatorConfig{ObjectsPerSeat: 100}
```

Comment:
Please define constants for all of those rather than hard-coding the values across the code.

Reply:
Done.